### PR TITLE
test(pkg/auth/providerrefresh): add tests to dameon suite

### PR DESCRIPTION
**Pull request by alegrey91**
## Solution
Added tests for several functions changed in the previous PR: https://github.com/rancher/rancher/pull/46814

### CODE REVIEW

File: pkg/auth/providerrefresh/refresher.go [hunk 1]
```diff
@@ -47,6 +48,7 @@ type refresher struct {
 	intervalInSeconds   int64
 	unparsedMaxAge      string
 	maxAge              time.Duration
+	cron                cron.Cron
```
Comment (bigkevmcd) [inline pkg/auth/providerrefresh/refresher.go:51]: I'd make this `*cron.Cron` which would let you use the return value from `cron.New()`?

File: pkg/auth/providerrefresh/daemon_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	ref := &refresher{}
+	ref.unparsedMaxAge = ""
+
+	err := ref.updateRefreshMaxAge("60")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if ref.maxAge != time.Duration(60*time.Second) {
+		t.Fatalf("error: attribute max age is wrong")
```
Comment (bigkevmcd) [inline pkg/auth/providerrefresh/daemon_test.go:44]: Maybe include the value of `ref.maxAge` ?

File: pkg/auth/providerrefresh/daemon_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	}
+	if ref.unparsedMaxAge != "60" {
+		t.Fatalf("error: unparsed max age is wrong")
```
Comment (bigkevmcd) [inline pkg/auth/providerrefresh/daemon_test.go:47]: Maybe include the value of `ref.unparsedMaxAge` ?

Whenever I use `Fatalf` without any interpolated values, I look for something that would help understand what went wrong 😄

File: pkg/auth/providerrefresh/daemon_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/robfig/cron"
+)
+
+func Test_updateRefreshCronTime(t *testing.T) {
+	ref := &refresher{}
+
+	ref.cron = *cron.New()
+	currentRefresh, err := cron.ParseStandard("0 0 * * *")
+	if err != nil {
+		t.Fatalf("error parsing cron time")
```
Comment (bigkevmcd) [inline pkg/auth/providerrefresh/daemon_test.go:17]: Maybe include the error?

Generally, for "unexpected errors" I tend to use `t.Fatal(err)`

File: pkg/auth/providerrefresh/daemon.go [hunk 5]
```diff
@@ -15,29 +14,36 @@ import (
...
 		userAttributeLister: mgmtContext.Management.UserAttributes("").Controller().Lister(),
+		cron:                *cron.New(),
 	}
 
 	UpdateRefreshMaxAge(maxAge)
 	UpdateRefreshCronTime(refreshCronTime)
 
 }
 
+// UpdateRefreshMaxAge updates the current refresh cron time with the one provided as input.
```
Comment (andreas-kupries) [inline pkg/auth/providerrefresh/daemon.go:38]:
```suggestion
// UpdateRefreshCronTime updates the current refresh cron time with the one provided as input.
```

File: pkg/auth/providerrefresh/daemon.go [hunk 6]
```diff
@@ -46,40 +52,49 @@ func UpdateRefreshCronTime(refreshCronTime string) error {
...
 		job := cron.FuncJob(RefreshAllForCron)
-		c.Schedule(parsed, job)
-		c.Start()
+		ref.cron.Schedule(parsed, job)
+		ref.cron.Start()
 	}
 	return nil
 }
 
+// UpdateRefreshMaxAge parse the maxAge string given in input and set
```
Comment (andreas-kupries) [inline pkg/auth/providerrefresh/daemon.go:66]:
```suggestion
// UpdateRefreshMaxAge parse the maxAge string given in input and sets
```

File: pkg/auth/providerrefresh/daemon.go [hunk 7]
```diff
@@ -46,40 +52,49 @@ func UpdateRefreshCronTime(refreshCronTime string) error {
...
+func (ref *refresher) updateRefreshMaxAge(maxAge string) error {
+	if maxAge == "" {
 		return fmt.Errorf("refresh max age must be provided")
 	}
 
 	ref.ensureMaxAgeUpToDate(maxAge)
 	return nil
 }
 
+// RefreshAllForCron refresh all the users crons.
```
Comment (andreas-kupries) [inline pkg/auth/providerrefresh/daemon.go:83]:
```suggestion
// RefreshAllForCron refreshes all the users crons.
```

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: pkg/auth/providerrefresh/daemon_test.go (hunk 4 above)
Comment (alegrey91) [reply pkg/auth/providerrefresh/daemon_test.go:17]: sure, thanks for pointing it out

**Review by pmatseykanets — COMMENTED** [review]
Some time ago we introduced a new package `pkg/crondaemon` to abstract the daemon part making it reusable.
It's worth considering switching provider refresh to this package instead since we're touching it's daemon code.

**Pull request closed without merging**